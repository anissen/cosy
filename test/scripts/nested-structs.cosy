{
    struct Sizes {
        mut shoe_size Num
        mut height Num
    }

    struct Person {
        var age Num
        mut name Str
        var sizes Sizes // TODO: This is immutable and should be const!
    }

    var john = Person {
        age = 42,
        name = 'John',
        sizes = Sizes {
            shoe_size = 33,
            height = 186
        }
    }
    print john // height: 186
    // john.sizes.height += 10 // TODO: This should work
    john.sizes.height = john.sizes.height + 10 // TODO: This should fail
    print john // height: 196

    mut peter = john
    peter.name = 'Peter'
    peter.sizes.height = peter.sizes.height + 10
    print peter // height: 206
    
    print john  // height: 206
}
