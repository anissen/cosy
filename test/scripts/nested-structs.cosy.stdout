Person { age = 42, name = "John", sizes = Sizes { height = 186, shoe_size = 33 } }
Person { age = 42, name = "John", sizes = Sizes { height = 196, shoe_size = 33 } }
Person { age = 42, name = "Peter", sizes = Sizes { height = 206, shoe_size = 33 } }
Person { age = 42, name = "Peter", sizes = Sizes { height = 206, shoe_size = 33 } }
